apps/web/app/modules/Admin/Users/Users.page.tsx

```diff
@@ -181,42 +236,58 @@ const Users = () => {
     });
   };
 
+  const handleUsersGroups = () => {
+    updateUsersGroups({
+      data: {
+        userIds: selectedUsers,
+        groupId: selectedValue,
+      },
+    }).then(() => {
+      table.resetRowSelection();
+      setShowEditModal(null);
+    });
+  };
+
   const handleRowClick = (userId: string) => {
     navigate(userId);
   };
 
+  const editMutation = {
+    role: () => {},
+    group: handleUsersGroups,
+    delete: handleDeleteUsers,
+  };
+
   return (
     <div className="flex flex-col">
-      <h4 className={"text-2xl font-bold"}>{t("navigationSideBar.users")}</h4>
+      {showEditModal && (
+        <EditModal
+          type={showEditModal}
+          onConfirm={editMutation[showEditModal]}
+          onCancel={() => setShowEditModal(null)}
+          groupData={groups}
+          roleData={[USER_ROLE.admin, USER_ROLE.student, USER_ROLE.contentCreator]}
+          selectedUsers={selectedUsers.length}
+          selectedValue={selectedValue}
+          setSelectedValue={setSelectedValue}
+        />
+      )}
+      <div className="flex justify-between">
+        <h4 className="text-2xl font-bold">{t("navigationSideBar.users")}</h4>
+        <div className="flex gap-3">
+          <Link to="new">
+            <Button variant="primary">{t("adminUsersView.button.createNew")}</Button>
+          </Link>
```

Comment:
Please use `asChild` with `<Button>` instead of nesting `<Button>` inside `<Link>`.
This approach avoids invalid HTML (`<a><button></button></a>`,  block inside inline) and ensures proper semantics.
```suggestion
         <Button variant="primary" asChild>
            <Link to="new">
              {t("adminUsersView.button.createNew")}
            </Link>
          </Button>
```